### PR TITLE
PlotSquared Events and minor changes

### PR DESCRIPTION
**Pull request by Mwthorn**
Added mainly PlotSquared events when a player leaves/enters a plot. Also changed some minor spelling errors on other supported stuff.

This build has been tested on a 1.8 server which works as expected but an error occured when testing on production server. This error is said to be corrected on newer spigot/denizen (1.12) versions so I guess this is fine.

Also sorry about the obscured commits, hope this can be combined to one. It can still merge :)

### CODE REVIEW

**Comment by mcmonkey4eva** [comment]
Hey! 
Could you please, in all the tags, change `plotsquaredplot@plot` to `plotsquaredplot@plotsquaredplot`
That way the bot can differentiate them, lol

File: bukkit/src/main/java/com/denizenscript/depenizen/bukkit/objects/dPlotSquaredPlot.java [hunk 1]
```diff
@@ -35,6 +36,7 @@ public static dPlotSquaredPlot valueOf(String string, TagContext context) {
         // Match plotsquaredplot name
 
         string = string.replace("plotsquaredplot@", "");
+        dB.log(string);
```
Comment (mcmonkey4eva) [inline bukkit/src/main/java/com/denizenscript/depenizen/bukkit/objects/dPlotSquaredPlot.java]: Wat ^

File: bukkit/src/main/java/com/denizenscript/depenizen/bukkit/objects/dPlotSquaredPlot.java [hunk 2]
```diff
@@ -76,6 +78,14 @@ public dPlotSquaredPlot setPrefix(String prefix) {
     }
 
     @Override
+    public boolean equals(Object a) {
+        if (a instanceof dPlotSquaredPlot) {
+            return ((dPlotSquaredPlot) a).plot.getId().equals(plot.getId());
+        }
+        return false;
+    }
```
Comment (Morphan1) [inline bukkit/src/main/java/com/denizenscript/depenizen/bukkit/objects/dPlotSquaredPlot.java]: You shouldn't override the `equals` method in Java without also overriding `hashCode`, it leads to some funky stuff. Let us know if you need help figuring that out

**Comment by Mwthorn** [comment]
I tried this build on a production server and for some reason, whenever I use the tag "as_plotsquared_plot" it sometimes end up being timed out, like at every second time its used. This is the debug: http://old.mcmonkey.org/paste/44099/page/2#264 Which shows a successful run and another being timed out.

This error was similiar in previous builds but with an exception instead of a timeout http://old.mcmonkey.org/paste/44101/page/2#182

File: bukkit/src/main/java/com/denizenscript/depenizen/bukkit/objects/dPlotSquaredPlot.java (hunk 2 above)
Comment (Mwthorn) [reply bukkit/src/main/java/com/denizenscript/depenizen/bukkit/objects/dPlotSquaredPlot.java]: I tried contacting you on Discord about this. I do need some help with this one.

**Comment by Mwthorn** [comment]
After some changes and tests, things are looking good now. The above errors have been corrected. Thanks to some devs at PlotSquared :)

This pull request is ready once again.

**Pull request merged**